### PR TITLE
Extend deck builder format

### PR DESCRIPTION
**Pull request by myangelkamikaze**

### CODE REVIEW

File: ElectronicObserver/Services/DataSerializationService.cs [hunk 1]
```diff
@@ -284,11 +296,83 @@ bool maxAircraftLevel
...
+			MapAreaId = sortieDetails.World,
+			MapInfoId = sortieDetails.Map,
+			Cells = sortieDetails.Nodes
+				.OfType<BattleNode>()
+				.Select(MakeDeckBuilderCell)
+				.ToList(),
+		},
+	};
+
+	private static DeckBuilderCell MakeDeckBuilderCell(BattleNode node) => new DeckBuilderCell
```
Comment (Jebzou) [inline ElectronicObserver/Services/DataSerializationService.cs:319]: can use new()

File: ElectronicObserver/Services/DataSerializationService.cs
Comment (Jebzou) [inline ElectronicObserver/Services/DataSerializationService.cs:1]: Could use a single object as parameter for the methods with more than 7 parameters to make the code easier to read and fix some sonars

Comment (myangelkamikaze) [reply ElectronicObserver/Services/DataSerializationService.cs:1]: Yeah, a class for all the parameters makes sense.

File: ElectronicObserver/Window/Tools/ExpeditionRecordViewer/ExpeditionRecordViewerViewModel.cs [hunk 2]
```diff
@@ -179,11 +179,11 @@ private async Task OpenFleetImageGenerator()
 
 		int hqLevel = await SelectedExpedition.Model.GetAdmiralLevel(Db) ?? KCDatabase.Instance.Admiral.Level;
 
-		DeckBuilderData data = DataSerializationService.MakeDeckBuilderData
-		(
-			hqLevel,
-			SelectedExpedition?.Model.Fleet.MakeFleet(0)
-		);
+		DeckBuilderData data = DataSerializationService.MakeDeckBuilderData(new()
```
Comment (Jebzou) [inline ElectronicObserver/Window/Tools/ExpeditionRecordViewer/ExpeditionRecordViewerViewModel.cs:180]: Property DataSerializationService isn't used anymore
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=421&id=ElectronicObserverEN_ElectronicObserver&open=AYwoupmhgoX7C9Oq2RRV

File: ElectronicObserver/Window/Tools/FleetImageGenerator/FleetImageGeneratorViewModel.cs
Comment (Jebzou) [inline ElectronicObserver/Window/Tools/FleetImageGenerator/FleetImageGeneratorViewModel.cs:1]: Property DataSerialization isn't used anymore 
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=421&id=ElectronicObserverEN_ElectronicObserver&open=AYwoupl9goX7C9Oq2RRT

File: ElectronicObserver/Services/DataSerializationService.cs [hunk 3]
```diff
@@ -7,11 +7,15 @@
...
 using ElectronicObserver.Window.Tools.EventLockPlanner;
+using ElectronicObserver.Window.Tools.SortieRecordViewer.Sortie.Battle.Phase;
+using ElectronicObserver.Window.Tools.SortieRecordViewer.Sortie.Node;
+using ElectronicObserver.Window.Tools.SortieRecordViewer.SortieDetail;
 using ElectronicObserverTypes;
 using ElectronicObserverTypes.Serialization.AirControlSimulator;
 using ElectronicObserverTypes.Serialization.DeckBuilder;
 using ElectronicObserverTypes.Serialization.EventLockPlanner;
 using ElectronicObserverTypes.Serialization.FleetAnalysis;
+using static System.Windows.Forms.VisualStyles.VisualStyleElement.ListView;
```
Comment (Jebzou) [inline ElectronicObserver/Services/DataSerializationService.cs:18]: using to remove

**Review by Jebzou — APPROVED** [review]

**Review by Jebzou — APPROVED** [review]

**Pull request merged**